### DISCUSSION TITLE
GoPro Hero 11 and USB Power Trigger

### DISCUSSION THREAD

**Discussion by oraclerouter** [gopro/labs#320] [General]
Hi,

I'm looking to setup the GoPro as a traditional dash cam: start recording on USB power, loop the recording and shut it down after USB power is off. There has been some discussion here that there is a new better way of doing this on the Hero 11 using the wake feature. Can someone provide an example on how this can be done via the "new" way?

**Comment by dnewman-gpsw** [comment]
I posted a video on this, https://www.youtube.com/watch?v=T87RekjaVA4

There is a lot you can do with new the HERO11 Labs command set.

Here is the code used in the demo video:
![DashCamQR](https://user-images.githubusercontent.com/10856084/234622922-2a15a831-a56c-43ae-8352-65ee7a50798e.png)

**Reply by oraclerouter** [reply]
Thanks @dnewman-gpsw  this helps.
As I'm trying to break down the command set can you tell me what does this do `!MBOOT="!Ldcam"!SAVEdcam ` Is it just a custom command name you gave it?

**Reply by oraclerouter** [reply]
Also, how can I loop it for 20 minutes instead of using up the entire SD card?

**Reply by dnewman-gpsw** [reply]
Just select loop mode as you default for video mode.   The code above doesn't set any mode, it even works in timelapse.   So add the loop preset to you camera, the same QR code will work.

Steps for adding loop mode on camera:
![image](https://user-images.githubusercontent.com/10856084/234973987-3d04da1f-bb8c-44db-a54f-4852c5cd8248.png)

**Reply by oraclerouter** [reply]
Thanks. As I'm trying to break down the command set can you tell me what does this do !MBOOT="!Ldcam"!SAVEdcam  Is it just a custom command name you gave it?  I can't find anywhere what Ldcam or  SAVEdcam do.

**Reply by dnewman-gpsw** [reply]
Docs on action commands here (commands that start with !): https://gopro.github.io/labs/control/actions/ and here for BOOT in particular https://gopro.github.io/labs/control/extensions/

**Reply by oraclerouter** [reply]
perfect. so far so good. the last question I have is there a way to specify in the command to always use the looping setting. There may be times where I take the gopro out to take normal videos and if I forget to select the looping  mode it will use the other last used mode.

**Reply by dnewman-gpsw** [reply]
fairly easy,  add mVL for mode Video Loop

!SAVEdcam=<u0!R105mVL!C16<u0<a0.05!R15=L+1>L99!C16+=L0=Bt:B>u0<r0!S+=AB>u0=AB>a0.05=AB=B-A>B10>r0<u0!E<r0>B30!O!R28!B2

These scripts are very flexible, I'm sure a lot better one will be created for this over time.

**Reply by unknownsolo** [reply]
Any chance you can simplify this by just doing if the GoPro is powered by USB, start in "dashcam" mode and start recording right away, without all the motion stuff, if not USB powered, start in manual mode?

(including the auto cleanup feature)

**Reply by dnewman-gpsw** [reply]
Yes, you can do that. Use: https://gopro.github.io/labs/build/ to create that type of script.

**Reply by unknownsolo** [reply]
I honestly have no idea how to work with this! I tried following some of the comments over here: https://github.com/gopro/labs/discussions/322, and the ones in this thread, and I am not able to get it to work.

**Reply by dnewman-gpsw** [reply]
Not too hard with any coding skills.  

Some starting points
!MWAKE=2    <- use this to make the camera boot on USB power.

![Basic DashCam-11_29_2023, 17_00_05](https://github.com/gopro/labs/assets/10856084/514fccdf-4b34-4176-9095-b80e22359fb9)

If you load the about image (the PNG file) into  https://gopro.github.io/labs/build/ it will show the code like this
![image](https://github.com/gopro/labs/assets/10856084/4e25aa97-f942-4c3a-879f-678bba0691d9)

Which you can edit to create whatever behavior you want.

**Reply by unknownsolo** [reply]
This is exactly what I needed - thank you! 

I've added one extra step to clear the SD space if full, and now the camera is frozen on boot. Is it because it's trying to calculate how much it needs to clear? I have a 128GB SD card, and it's pretty much new and empty. It only has about 2GBs used. I formatted it from the camera, too. 

![Screenshot 2023-11-29 at 6 23 05 PM](https://github.com/gopro/labs/assets/86228066/a6aa6b46-ea2e-4ab6-96f9-9991b56608e5)

![IMG_3280](https://github.com/gopro/labs/assets/86228066/6e6931c6-2d2f-47d9-b8d3-22e4b3fe9fd4)

**Reply by dnewman-gpsw** [reply]
The boot command is on the sd card, so just eject the SD and restart the camera, it will not run the script. But I don't think is the issue, as the same command works here.  Factory reset your camera, and try again.

**Reply by unknownsolo** [reply]
I think a factory reset fixed it! It's now working as it should. One last question: Are the sleep steps at the bottom necessary? Is it to give the camera time to save the video before shutting down? 

Also, is there a way for me to contribute to your time and your help? I appreciate you!

**Reply by dnewman-gpsw** [reply]
The last one is a just in case.  The second last is completely unnecessary.

**Reply by dnewman-gpsw** [reply]
Contribute through telling others of the power of Labs.

**Reply by unknownsolo** [reply]
Will do :) Thanks again.

**Reply by Humanfuse** [reply]
I'm trying to get my Hero 11 FW V2.20.70 working with this method.  If I use the script on the top of this thread, the camera wakes up on USB power but I can't get it to start recording.   I assume capture was not getting triggered??
I tried the simple version below and now the camera will not wake up.  I'm using !RESET in between tries.  A simple !MWAKE=2 comand get it to wake up but not record?  Thoughts on what I'm doing wrong?
![Screenshot 2023-12-31 170451](https://github.com/gopro/labs/assets/155323914/3aed9268-fb3b-4b00-beda-57f42ff67423)

!MBOOT="!Lbt"!SAVEbt=<u0!X"Jer's Cam">u0<r0!S<u0>r0!5NQ+!E+!3N+!1O!R16

qr.bin file: 
<u0!X"Jer's Cam">u0<r0!S<u0>r0!5NQ+!E+!3N+!1O!R16

qrlog file:
[2023-12-31 19:03:08] Reset:QR Data 
[2023-12-31 19:04:10-delayed] wake reason:POWER_BUTTON 
[2023-12-31 19:04:11] Labs:Display Messages 
[2023-12-31 19:04:13] Labs:Main Start 
[2023-12-31 19:04:14] Labs:H22.01.02.20.70 
[2023-12-31 19:04:14] Labs:Initialize 
[2023-12-31 19:07:24-delayed] wake reason:POWER_BUTTON 
[2023-12-31 19:07:24] Labs:Display Messages 
[2023-12-31 19:07:26] Labs:Main Start 
[2023-12-31 19:07:26] Labs:H22.01.02.20.70 
[2023-12-31 19:07:26] Labs:Initialize 
[2023-12-31 19:07:30] Read QR:!MWAKE=2 
[2023-12-31 19:07:30] Read QR:!MWAKE=2 
[2023-12-31 19:07:31] Read QR:!MWAKE=2 
[2023-12-31 19:07:31] Read QR:!MWAKE=2 
[2023-12-31 19:07:31] Read QR:!MWAKE=2 
[2023-12-31 19:07:31] Read QR:!MWAKE=2 
[2023-12-31 19:07:57-delayed] wake reason:USB1_INSERTED 
[2023-12-31 19:07:59] Labs:Display Messages 
[2023-12-31 19:08:03] Features activated: 
WAKE=2: 
[2023-12-31 19:08:03] Labs:Main Start 
[2023-12-31 19:08:03] Labs:H22.01.02.20.70 
[2023-12-31 19:08:03] Labs:Initialize 
[2023-12-31 19:08:19] Read QR:!MBOOT="!Lbt"!SAVEbt=<u0!X"Jer's Cam">u0<r0!S<u0>r0!5NQ+!E+!3N+!1O!R16 
[2023-12-31 19:08:20] Save Script:C:\MISC\bt 
[2023-12-31 19:08:21] Boot command:!Lbt 
[2023-12-31 19:08:23] Load Script:C:\MISC\bt 
[2023-12-31 19:08:43-delayed] wake reason:USB1_INSERTED 
[2023-12-31 19:08:45] Labs:Display Messages 
[2023-12-31 19:08:49] Features activated: 
WAKE=2,BOOT: 
[2023-12-31 19:08:49] Labs:Main Start 
[2023-12-31 19:08:49] Labs:H22.01.02.20.70 
[2023-12-31 19:08:49] Labs:Initialize 
[2023-12-31 19:08:50] Boot command:!Lbt 
[2023-12-31 19:08:51] Load Script:C:\MISC\bt

**Reply by dnewman-gpsw** [reply]
The visual Blockly code and the command string are not the same.  Did you incorrectly transcribe or was the Blockly generation wrong?

This works:
!MWAKE=2!MBOOT="!Lbt"!SAVEbt=<u0!X"Jer's Cam">u0<r0!S<u0>r0!5NQ+!E+!3N+!1O!R16

Select **Save/Generator QR**, and then **Download QR**.  Directly share that downloaded .png file here.

**Reply by Humanfuse** [reply]
Thank you for the reply!  The block code does not match because I first manually entered a !MWAKE=2 command, then I sent the QR code, !MBOOT="!Lbt"!SAVEbt=>u0<r0!S<u0>r0!5NQ+!E+!3N+!1O!R  That did not work.  Wakes but does not record.  It never displays "Jerry Cam".  From the log, it looks like the script loads but does not run??

I tried your code and get the same result.  It wakes up but will not capture.  

qr.bin
<u0!X"Jerry's Cam">u0<r0!Sr0!5NQ+!E+!3N+!1O!R16

Log:  3 boots
[2023-12-31 19:52:30] Labs:Main Start 
[2023-12-31 19:52:31] Labs:H22.01.02.20.70 
[2023-12-31 19:52:31] Labs:Initialize 
[2023-12-31 19:52:31] Read QR:!MWAKE=2!MBOOT="!Lbt"!SAVEbt=<u0!X"Jerry's Cam">u0<r0!Sr0!5NQ+!E+!3N+!1O!R16 
[2023-12-31 19:52:31] Save Script:C:\MISC\bt 
[2023-12-31 19:52:33] Boot command:!Lbt 
[2023-12-31 19:52:33] Load Script:C:\MISC\bt 

[2023-12-31 19:53:43-delayed] wake reason:USB1_INSERTED 
[2023-12-31 19:53:45] Labs:Display Messages 
[2023-12-31 19:53:49] Features activated: WAKE=2,BOOT: 
[2023-12-31 19:53:49] Labs:Main Start 
[2023-12-31 19:53:49] Labs:H22.01.02.20.70 
[2023-12-31 19:53:49] Labs:Initialize 
[2023-12-31 19:53:50] Boot command:!Lbt 
[2023-12-31 19:53:51] Load Script:C:\MISC\bt 

[2023-12-31 19:55:04-delayed] wake reason:USB1_INSERTED 
[2023-12-31 19:55:06] Labs:Display Messages 
[2023-12-31 19:55:10] Features activated: WAKE=2,BOOT: 
[2023-12-31 19:55:10] Labs:Main Start 
[2023-12-31 19:55:10] Labs:H22.01.02.20.70 
[2023-12-31 19:55:10] Labs:Initialize 
[2023-12-31 19:55:11] Boot command:!Lbt 
[2023-12-31 19:55:12] Load Script:C:\MISC\bt 

[2023-12-31 19:56:57-delayed] wake reason:USB1_INSERTED 
[2023-12-31 19:56:59] Labs:Display Messages 
[2023-12-31 19:57:03] Features activated: WAKE=2,BOOT: 
[2023-12-31 19:57:03] Labs:Main Start 
[2023-12-31 19:57:03] Labs:H22.01.02.20.70 
[2023-12-31 19:57:03] Labs:Initialize 
[2023-12-31 19:57:04] Boot command:!Lbt 
[2023-12-31 19:57:05] Load Script:C:\MISC\bt

**Reply by dnewman-gpsw** [reply]
The code you are running is wrong, but I see that is truncated when posted as characters are missing, so it is the forum's bug.  The wrong code is in the log.  So share the png, we can fix from there.

**Reply by Humanfuse** [reply]
Happy New Year!  To be clear, I copy/pasted the code you suggested:  !MWAKE=2!MBOOT="!Lbt"!SAVEbt=<u0!X"Jer's Cam">u0<r0!Sr0!5NQ+!E+!3N+!1O!R16 into the GoPro QR Code Creator.  I do not have a png image for it but here is a screen shot.

![Screenshot 2024-01-01 003215](https://github.com/gopro/labs/assets/155323914/4be4dbe0-4094-46db-99e0-60894489f9b3)

If I go back to my block code from my first post, I'm stuck because I seem to need the !MWAKE=2 before I send rest.  How do I add the wake command to the block code?  Here is my current block code and png. It does not wake the camera with USB.

![Screenshot 2024-01-01 004346](https://github.com/gopro/labs/assets/155323914/f1bee83a-9845-46b1-ab3b-6cadfbc083de)

![New Project-1_1_2024, 00 47 42](https://github.com/gopro/labs/assets/155323914/20be4200-473e-49ce-8660-1289f808e7ce)

Log
[2024-01-01 02:01:00] Reset:QR Data 
[2024-01-01 02:01:24] Read QR:!MBOOT="!Lbt"!SAVEbt=<u0!X"Cam Test">u0<r0!S<u0>r0!5NQ+!E+!3N+!1O!R15 
[2024-01-01 02:01:24] Save Script:C:\MISC\bt 
[2024-01-01 02:01:25] Boot command:!Lbt 
[2024-01-01 02:01:26] Load Script:C:\MISC\bt 

Edit:  I think I see the camera briefly turn on (screen back light only/no text) then back off with this code.

**Reply by dnewman-gpsw** [reply]
The copy/pasted code is wrong as the forum is dropping characters. The code directly from blockly is correct. Look closely are you will they are different. If you restore the missing characters it will work in web tool.  !MWAKE=2 is only needed once after a !RESET.

**Reply by Humanfuse** [reply]
hummm.  OK, Thank you, I get that there is a copy/paste issue with the forum.  Unfortunately, I'm still not finding the issue with my blockly code?  

As an experiment, I tried a work around but I'd like to get the blockly method working too.  

Using the web tool here: https://gopro.github.io/labs/control/custom/ I sent the camera a !RESET then a !MWAKE=2
That part works to turn on the camera with USB power.  Then I sent the camera the IMU Trigger command !SI6D1H5!R to capture with camera movement.

That works.  It turns on the cam with USB power, stops/starts record if cam is moving and then I just let the camera auto power down.  Not ideal for the car but I think that proves my USB power is OK, my labs FW was loaded correctly, script is loading, etc.

You state my blockly code looks correct.  Great.  When I generate the QR code I get 
!MBOOT="!Lbt"!SAVEbt=<u0!X"Cam Test">u0<r0!S<u0>r0!5NQ+!E+!3N+!1O!R15

![New Project-1_1_2024, 04 07 20](https://github.com/gopro/labs/assets/155323914/a0877d00-a6ee-4115-884d-6597e96a14ba)

I manually sent the H11 a reset, then a wake with web tool again, then I scanned the above Blockly QR code.  Camera is waking up but not recording?  While booting, display reads "Feature Activated, WAKE 2, BOOT.  Then it displays "Boot command: !Lbt" After boot, I'm getting a new small green square in the upper left corner of the camera icon.  

I get simular log result.  Script loads but does not appear to run??  What the heck am I missing??  Thoughts?

Log[2024-01-01 04:27:12] Features activated: 
WAKE=2,BOOT: 
[2024-01-01 04:27:12] Labs:Main Start 
[2024-01-01 04:27:12] Labs:H22.01.02.20.70 
[2024-01-01 04:27:12] Labs:Initialize 
[2024-01-01 04:27:13] Boot command:!Lbt 
[2024-01-01 04:27:14] Load Script:C:\MISC\bt

**Reply by Humanfuse** [reply]
Please disregard the above post.  I did not read enough.  The issue turned out to be that I was using a 2.0 amp wall wart to trigger the camera. (Requires 2.4 amp) It would wake on a simple !MWAKE=2 but not on other harder scripts.  I was tricked, LOL.  I changed to powering from a 2.4 amp outlet on my power brick.  All started working as expected.  <smh>  Now I can dive in to deeper Blockly code knowing that it's working as expected. 

I moved my now working H11 Dash Cam to my truck for a test.  It has fancy USB A and C jacks on the dash and center compartment but they are all hooked to truck's USB data as well as power.  The H11 launches into USB connected mode.  It switches back to the camera screen after a few seconds but will not automatically record.  If I go back to testing using my USB power brick, everything works as expected.  USB connected mode seems to interfere with the script??

Is there a command to ignore USB data connection?  I know about TUSB=1 for low amps supplies but that is not my problem.  As a workaround, I'll order a "power only" USB cable and see if that works.  Thoughts on a more elegant solution?

Thank you to you and the Labs team.  Info in this forum is very helpful.

Edit, I have a USB-C "Data Blocker" on the way.  I found this is a common problem on newer vehicles with Factory USB data.
https://www.youtube.com/watch?v=CISXQoFvfH0

**Reply by dnewman-gpsw** [reply]
Happy to hear you are making progress

This is correct:
```
!MWAKE=2!MBOOT="!Lbt"!SAVEbt=<u0!X"Jer's Cam">u0<r0!S<u0>r0!5NQ+!E+!3N+!1O!R16
```

For some reason the forum removes 
```
<u0>
```

from the script if the code is not embedded in a codeblock.  It is an important element.

**Reply by unknownsolo** [reply]
@dnewman-gpsw - Is there a way to add !MWAKE=2 in blockly?

**Reply by dnewman-gpsw** [reply]
Any custom command can be added like this:

![image](https://github.com/gopro/labs/assets/10856084/76778af8-ca12-42c4-8210-8158de1f248e)

Previously boot commands didn't support an input block, but that has just been fixed.  You may need to refresh Blockly.

**Reply by Humanfuse** [reply]
To be safe, I added the !MWAKE=2 command to my code.  Works as expected.

Thank you very much!!!

![Screenshot 2024-01-01 161157](https://github.com/gopro/labs/assets/155323914/da9072d2-3955-4849-b0eb-7b8cc6500513)

**Reply by unknownsolo** [reply]
Sweet! Thank you! 

Now we're talking! I added MWAKE and MFAST. 

![Screenshot 2024-01-01 at 1 20 24 PM](https://github.com/gopro/labs/assets/86228066/f994f4c9-e9f7-4f43-87b6-fdbb0a644927)

**Reply by Humanfuse** [reply]
Nice.  I see the battery level check and SD card clean up.   I added a few print to screen steps to help see the code run. Now I'm looking to send setting for the right compromise of vid quality vs file size.  I'm getting 100 MB/20 sec MP4 files on 1080 with 60 fps.  Decent video quality.  I have a 256GB card so I assume that about ~40 minutes.  Sounds reasonable with the looping preset enabled.   Thoughts?

**Reply by unknownsolo** [reply]
I started with 4K, then switched to 1080p with 60 fps. 1080p with 60fps is more than enough for a dash cam for now. I also have a 256GB card. I set my loop to 20 minutes just in case.

**Reply by unknownsolo** [reply]
@dnewman-gpsw - I found a bug with the QR generator in Blockly. I've noticed that my script doesn't work right when I add the custom QR command at the very top. When I removed it, everything worked without issues. When I scan the QR with my iPhone camera, the output is different from what is printed in the output image:

![Dash Cam-1_7_2024, 16_19_44](https://github.com/gopro/labs/assets/86228066/d0e764e0-0b38-49a1-bb6f-96cd1a5a085c)

![Screenshot 2024-01-07 at 4 21 31 PM](https://github.com/gopro/labs/assets/86228066/e790142b-9efd-4934-b7fa-180e3c3cf455)

If you scan the above image, the last character is a 2, which should be a 3 in the picture. 

QR Image: !MWAKE=2!MFAST=1!MBOOT="! Lbt" ! SAVEbt=<u0!X"Dash Cam">u0>b80<r0!C16+!S<u0!E+!1O!R5**3**
Phone QR scan: !MWAKE=2!MFAST=1!MBOOT="! Lbt" ! SAVEbt=<u0!X"Dash Cam">u0>b80<r0!C16+!S<u0!E+!1O!R5**2**

**Reply by dnewman-gpsw** [reply]
Okay, I'll look into that, as neither !R52 or !R53 are correct, should be !R15.  The code before boot with is confusing the character counter.

**Reply by unknownsolo** [reply]
Thank you! No biggie, I am doing !MWAKE separately, and it's working great so far. I just thought I'd let you know.

**Reply by dnewman-gpsw** [reply]
Now fixed.

**Reply by unknownsolo** [reply]
Thank you!

**Reply by pairahoos** [reply]
I've managed to load one of these scripts (first time playing with this) and it does exactly what I've needed, but I have no idea how to stop it. I've done a !RESET and a !FRESET and this script is still launching directly into video capture at power on. I've got a nice endless loop here. Wondering how I stop it!

**Reply by dnewman-gpsw** [reply]
Eject the SD Card. The script runs from the SD card.  The !RESET will also work, but you have to follow the onscreen instructions, pressing the shutter three time quickly, otherwise it does nothing.

**Reply by pairahoos** [reply]
That did the trick. Thank you. I believe I did the three "button" presses at start up with !RESET - and that did not work. Thanks for clarifying its the shutter "button." I wasn't sure which of the two and probably tried both.

**Reply by dthiss** [reply]
> I posted a video on this, https://www.youtube.com/watch?v=T87RekjaVA4
> 
> There is a lot you can do with new the HERO11 Labs command set.
> 
> Here is the code used in the demo video: ![DashCamQR](https://user-images.githubusercontent.com/10856084/234622922-2a15a831-a56c-43ae-8352-65ee7a50798e.png)

@dnewman-gpsw
I've been using this QR for a couple of weeks now.  Works well, I appreciate all of your posts.  I tried to make my own that had a way (2 presses of the shutter button) to escape out of the program but can't seem to make it work.  Is it possible for you to modify this and add an escape route?

**Reply by dthiss** [reply]
> > I posted a video on this, https://www.youtube.com/watch?v=T87RekjaVA4
> > There is a lot you can do with new the HERO11 Labs command set.
> > Here is the code used in the demo video: ![DashCamQR](https://user-images.githubusercontent.com/10856084/234622922-2a15a831-a56c-43ae-8352-65ee7a50798e.png)
> 
> @dnewman-gpsw I've been using this QR for a couple of weeks now. Works well, I appreciate all of your posts. I tried to make my own that had a way (2 presses of the shutter button) to escape out of the program but can't seem to make it work. Is it possible for you to modify this and add an escape route?

This is what I used, but doesn't seem to work.  Can you look it over to see if I have a mistake somewhere?

![image](https://github.com/gopro/labs/assets/170752725/f0187f4d-ea38-4816-93ac-881a0f29f1c9)

**Reply by dnewman-gpsw** [reply]
Post these question in a new thread.  This is too easy to get lost on the very long discussion.  I don't immediately see the issue, but you could simply further to test only the button cancel.

**Comment by RuiPereira** [comment]
Hi. I've been using this dashcam script derived from the one initially proposed by @dnewman-gpsw :

`<u0!R68<a0.05!R7!C32=At:B>u0<r0!S>r0<u0!10N+!E+!1O=A%1800<A2!C32!R20!B2`

It works fine most of the time, but sometimes it stops recording and restarts again right after.
The problem is random and does not seems to be correlated with available space (I've a 64GB card, clearing it every 30min), nor USB power stability, nor record time... 

Cam info:
> "info version":"2.0"
"firmware version":"H22.03.02.30.70"
"camera type":"HERO11 Black Mini"

Any thoughts? Is there a log command that I could use to try to identify a possible culprit? 
Here's a sample error log, where it bugged twice (18:29 / 18:54):

> [2023-11-29 18:27:51-delayed] wake reason:USB1_INSERTED 
[2023-11-29 18:27:51-delayed] Labs:Display Messages 
[2023-11-29 18:27:53] Labs:Main Start 
[2023-11-29 18:27:54] Labs:Initialize 
[2023-11-29 18:27:54] Boot command:!Ldcam 
[2023-11-29 18:27:55] Load Script:C:\MISC\dcam 
[2023-11-29 18:27:58] total size (GB):59 
[2023-11-29 18:27:58] remaining space (GB):30 
[2023-11-29 18:29:37] capture stopped:user input or error 
[2023-11-29 18:54:32] capture stopped:user input or error 
[2023-11-29 18:54:32] capture stopped:user input or error 
[2023-11-29 18:54:32] capture stopped:user input or error 
[2023-11-29 18:54:32] capture stopped:user input or error 
[2023-11-29 18:54:32] capture stopped:user input or error 
[2023-11-29 18:54:32] capture stopped:user input or error 
[2023-11-29 18:54:32] capture stopped:user input or error 
[2023-11-29 18:54:32] capture stopped:user input or error 
[2023-11-29 18:54:32] capture stopped:user input or error 
[2023-11-29 18:54:32] capture stopped:user input or error 
[2023-11-29 18:54:32] capture stopped:user input or error 
[2023-11-29 18:54:32] capture stopped:user input or error 
[2023-11-29 18:54:32] capture stopped:user input or error 
[2023-11-29 18:54:32] capture stopped:user input or error 
[2023-11-29 18:54:32] capture stopped:user input or error 
[2023-11-29 18:54:32] capture stopped:user input or error 
[2023-11-29 18:54:32] capture stopped:user input or error 
[2023-11-29 18:54:32] capture stopped:user input or error 
[2023-11-29 18:54:32] capture stopped:user input or error 
[2023-11-29 18:54:32] capture stopped:user input or error 
[2023-11-29 18:57:50] total size (GB):59 
[2023-11-29 18:57:50] remaining space (GB):21 
[2023-11-29 18:57:50] clear up to (GB):29 
[2023-11-29 18:57:50] Remove File:GL010310.LRV 
[2023-11-29 18:57:50] Remove File:GX010310.THM 
[2023-11-29 18:57:50] Remove File:GX010311.MP4 
[2023-11-29 18:57:50] Remove File:GL010311.LRV 
[2023-11-29 18:57:50] Remove File:GX010311.THM 
[2023-11-29 18:57:50] Remove File:GX010312.MP4 
[2023-11-29 18:57:50] Remove File:GL010312.LRV 
[2023-11-29 18:57:51] Remove File:GX010312.THM 
[2023-11-29 18:57:51] Remove File:GX010313.MP4 
[2023-11-29 18:57:51] Remove File:GL010313.LRV 
[2023-11-29 18:57:51] Remove File:GX010313.THM 
[2023-11-29 18:57:51] Remove File:GX010314.MP4 
[2023-11-29 18:57:51] new remainning space (GB):34 
[2023-11-29 18:57:52] total size (GB):59 
[2023-11-29 18:57:52] remaining space (GB):34 
[2023-11-29 18:57:52] total size (GB):59 
[2023-11-29 18:57:52] remaining space (GB):34 
[2023-11-29 18:57:52] total size (GB):59 
[2023-11-29 18:57:52] remaining space (GB):34 
[2023-11-29 19:02:19] delay:end 

Thanks!

**Reply by dnewman-gpsw** [reply]
The error happened externally to Labs, so the log is not showing much.  

[2023-11-29 18:29:37] capture stopped:user input or error 

This means that the system stopped the capture for reasons not listed. Are you over-cranking the bitrate or is the camera getting very hot?  Run some experiments at 1080p with a lower bitrate to see if it is the cause.

**Reply by RuiPereira** [reply]
I’m using 1080p at 30fps. I’ll try to lower it to 24fps to see if that changes anything. Thanks!

**Reply by RuiPereira** [reply]
Hi @dnewman-gpsw, it’s been a while, but I managed to grab an extended log of an error event. 

[qrlog.txt](https://github.com/gopro/labs/files/14440681/qrlog.txt)

Same firmware (latest .40.70 makes my camera crash while in standby during the night) and slightly different script, but right before the error there’s a line I’d never seen:

`[2024-02-27 18:55:30] Battery Temp.:20 `

Any idea?

**Reply by dnewman-gpsw** [reply]
That is not a crash, that is the battery temperature, it reports that occasionally.  Not related.

Where is the crash, is the camera just stop at 18:55:44?  Was the failure?

**Reply by RuiPereira** [reply]
The failure is at 18:55:31, when the camera beeps and stops recording. It restarts once again right after, before I disconnected the USB cable to shut it down.

**Reply by dnewman-gpsw** [reply]
The message is that something external happened, the SD errored, or user button press, or some other error.  It is not typically a sign of a crash, and the camera did not reboot.  Interesting that the problem was only temporary and the capture restarted.  Nothing in the Labs log reports the external actions.  Do this happen often?

**Reply by RuiPereira** [reply]
Hard to tell, but at least a couple times per week. No button clicked, and I’m using 1080p 24fps so no bandwidth problem I guess. I use the camera in “normal” (no scripting) mode for MTB and it records continuously for 2h without a hitch.

**Comment by AlpDnkts** [comment]
How to remove last update please i lost usb recording aftet update to last firmware:(

**Reply by dnewman-gpsw** [reply]
I expect you didn't, the feature just changed. USB power trigger got better. Which camera and firmware?

**Reply by AlpDnkts** [reply]
11 black,20.20 firmware, 
2 days i used with powerbank, evening time i updated something from app, after no work with usb, so annyoying

**Reply by AlpDnkts** [reply]
02.20.70 labs

**Reply by dnewman-gpsw** [reply]
Works fine with my HERO11s.

Try this. 
![USB Power Video Trigger-12_11_2023, 09_19_37](https://github.com/gopro/labs/assets/10856084/a3215954-f484-410d-8d0c-dc967bda4097)

**Reply by dnewman-gpsw** [reply]
This is combined with !MWAKE=2.  While the setup to more complex than the old power trigger, it way more reliable (and more flexible.)

**Reply by AlpDnkts** [reply]
how i can set up or downloand this?

**Reply by AlpDnkts** [reply]
when i scan with gopro this qr code, it says boot command!lbs

**Reply by AlpDnkts** [reply]
![image](https://github.com/gopro/labs/assets/153510156/30b6f71b-a314-4b94-b94b-f2391c84571b)

**Reply by AlpDnkts** [reply]
> ![image](https://github.com/gopro/labs/assets/153510156/30b6f71b-a314-4b94-b94b-f2391c84571b)

and still its not working with usb powerbank

**Reply by AlpDnkts** [reply]
oh my god dashcam qr code worked thanks alot

**Reply by AlpDnkts** [reply]
when i connect usb to gopro, it starts recording with standart mode and i couldnt change it to activity 2.7k 60

**Reply by dnewman-gpsw** [reply]
You could change standard or change the script to select wherever preset you want.

**Reply by AlpDnkts** [reply]
its not allowing me from screen touching, i dont know how i can do

**Reply by AlpDnkts** [reply]
and 15min recording after camera heating

**Reply by dnewman-gpsw** [reply]
Maybe you need to start your own thread, with what you want to do, and you what you have tried.  Labs is for experimentation, with millions of combinations and programmable scripting. It does not cause over heating, as it is uses the same video modes on the stock camera, and it does not disable the touchscreen.

**Reply by AlpDnkts** [reply]
dashcam script is not allowing me to change anything, plug in usb- start record, thats it, no more

**Reply by AlpDnkts** [reply]
I built a new type of settings with qr, and it worked, suprised😅

**Comment by ichibanh** [comment]
Would this work on a Hero 8 Black?

**Reply by dnewman-gpsw** [reply]
Not all of it, no.  HERO8 supports !MWAKE=2 (wake on power), and the older style dashcam mode  https://gopro.github.io/labs/control/usb/ , but it doesn't support to new custom scripting.

**Comment by SpenHub** [comment]
> I posted a video on this, https://www.youtube.com/watch?v=T87RekjaVA4
> 
> 
> 
> There is a lot you can do with new the HERO11 Labs command set.
> 
> 
> 
> Here is the code used in the demo video:
> 
> ![DashCamQR](https://user-images.githubusercontent.com/10856084/234622922-2a15a831-a56c-43ae-8352-65ee7a50798e.png)
> 
> 

Super late to the party. You ever figure out how to use the dash cam with a loop mode/ preset longer than 5 min?

I have the code mVL which sets video mode to current loop preset, but no matter what... it always sets loop duration to 5 min despite preset having it set to 20. Any ideas?

**Reply by dnewman-gpsw** [reply]
mVL just selects the preset, it will not change any settings.  A 20 loop, with keep four 5 minute chapters.  That is how it works.

**Reply by SpenHub** [reply]
I wish it was but it's not the case. 

To reiterate I have a preset mode called Looping with the loop preset set to 20min. When I use the dashcam script with the addition of mVL, I see odd behavior. 

I'll plug in the camera, it boots, records, I step away from 20+ min and see the camera's recording indicator stuck at 05:00 meaning it's saving a 5 minute loop. 

I end the recording, and check the files on the card, I'll only see 5 1 minute clips. It seems to not be using my looping preset.

**Reply by dnewman-gpsw** [reply]
What camera?  I tested on my HERO13 and it worked as I described.

**Reply by SpenHub** [reply]
I've got a hero 13 as well with latest firmware. (02.02.70) Here's some screenshots. 

First: showing the preset at the bottom ![image](https://github.com/user-attachments/assets/925fc800-d179-4867-a4ca-1104ba8d80db)

Second: showing the single 5 minute clip from one ride (from destination) ![image](https://github.com/user-attachments/assets/662f42b6-79c9-4b14-b9a2-0baf10f3e40e)

Third: showing the single 5 minute clip from one ride (to destination) ![image](https://github.com/user-attachments/assets/02654800-12f5-453d-bf1a-b6c8d24c9464)

**Comment by dnewman-gpsw** [comment]
Yes, correct, that is how it works. It will keep up to 20 minutes in 5 minute segments. Nothing to do with Labs, that is a stock camera feature.  The on camera playback of looping is weird, but all the segments are on the SD Card.

**Reply by SpenHub** [reply]
Aha! You're right. I didn't even think there's a diff between what the camera shows vs what the media on the card shows. I'd assume they're the same. Guess not. ![image](https://github.com/user-attachments/assets/86477183-7fcf-4e8b-a0a0-bcbafb085cd1)

Thank you so much. My ride this weekend was saved after all! Snippet as a thank you ❤️
https://github.com/user-attachments/assets/23f3d55f-b170-4557-8a4d-2957cb8a1821
